src/core/thread/mle.cpp

```diff
@@ -1055,6 +1058,16 @@ void Mle::InitNeighbor(Neighbor &aNeighbor, const RxInfo &aRxInfo)
     aNeighbor.SetLastHeard(TimerMilli::GetNow());
 }
 
+void Mle::ScheduleChildUpdateRequestIfMtdChild(void)
+{
+    VerifyOrExit(IsChild() && !IsFullThreadDevice());
```

Comment:
Since MTD is statically configured, what about we wrap these code with `OPENTHREAD_MTD` and we don't need call `IsFullThreadDevice` here. And we can save the unnecessary code in FTD.

Or do we actually mean MED here? Because we usually use FTD/MTD to indicate the static capability and we use FED/MED to indicate the run-time node type.

Reply:
> Since MTD is statically configured, what about we wrap these code with OPENTHREAD_MTD and we don't need call IsFullThreadDevice here. And we can save the unnecessary code in FTD.

This will not work and would not be correct.

We can have a device that's running a build with `OPENTHREAD_FTD` which is then configured to act as an MTD and would need to run this code.

---

> Or do we actually mean MED here?

We mean MED here. I will rename the method to make this clear.

Reply:
> Or do we actually mean MED here?

Actually, this situation is applicable to SED as well as MED. Per the Thread spec, an MED is an MTD child whose receive is always enabled.

I prefer to keep the term `"MtdChild"` for this, so to refer to any MTD child, whether it is SED or MED, etc. This helps keep it aligned with the `"FtdChild"` term which we use within OT code.

Earlier we had ambiguity in how we used the "FED" term within the OT code.
- Within the Thread spec definition, an FED is an FTD that acts as an ED and will **always** remain an ED and **never** request to become a router, unlike a REED.
- Within OT code, we used "Fed" in some places to refer to an "FTD child" (which can be FED or REED).
- Recently fixed this to use `"FtdChild"` within the code (https://github.com/openthread/openthread/pull/10718)).

Reply:
Thanks! That makes sense to me.